BiometricLoginKotlin/app/src/main/java/com/example/biometricloginsample/LoginWithPasswordFormState.kt

```diff
@@ -0,0 +1,23 @@
+
+/*
+ * Copyright (C) 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.biometricloginsample
+
+data class LoginWithPasswordFormState (
```

Comment:
At first glance, it looks like this class is doing two different things?

- Reporting errors for username/password
- Reporting successful login

Is that the case? If so, I'd recommend splitting this into two different classes and using a sealed class approach. Something like `LoginFormState` with `FailedLoginFormState` and `SuccessfulLoginFormState`, etc...